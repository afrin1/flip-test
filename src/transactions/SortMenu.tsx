import React, { useEffect, useState } from 'react';
import { Modal, Pressable, StyleSheet, Text, View } from 'react-native';
import { RadioButton } from '../components/Button';
import { RadioItem } from '../components/Button/Radio';

interface SortMenuProps {
  setMenuClose: any,
  onOptionSelection: any,
  selectedOption: string
}

const menuItems: RadioItem[] = [
  {
    key: 'URUTKAN',
    text: 'URUTKAN'
  },
  {
    key: 'Nama A-Z',
    text: 'Nama A-Z'
  },
  {
    key: 'Nama Z-A',
    text: 'Nama Z-A'
  },
  {
    key: 'Tanggal Terbaru',
    text: 'Tanggal Terbaru'
  },
  {
    key: 'Tanggal Terlama',
    text: 'Tanggal Terlama'
  },
]

const SortMenu = (props: SortMenuProps) => {
  const { setMenuClose, onOptionSelection, selectedOption } = props
  const [isOpen, setIsOpen] = useState<boolean>(true)

  const onClose = () => {
    setIsOpen(false)
    setMenuClose()
  }

  useEffect(() => {
    setIsOpen(true)
  }, [])
  return (
    <Modal
      animationType="slide"
      visible={isOpen}
      transparent={true}
      onRequestClose={onClose}
    >
      <Pressable
        style={styles.container}
        onPress={onClose}>
        <View style={styles.centeredView}>
          <RadioButton
            items={menuItems}
            defaultValue={selectedOption}
            onSelection={onOptionSelection} />
        </View>
      </Pressable>
    </Modal>
  )
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: 'rgba(0,0,0,0.5)'
  },
  centeredView: {
    justifyContent: "center",
    backgroundColor: 'white',
    padding: 10,
    borderRadius: 5,
    opacity: 1,
    flexGrow: 1,
    margin: 20
  },
  button: {
    borderRadius: 20,
    padding: 10,
    elevation: 2
  },
  buttonClose: {
    backgroundColor: "#2196F3",
  },
  textStyle: {
    color: "white",
    fontWeight: "bold",
    textAlign: "center"
  },
  modalText: {
    marginBottom: 15,
    textAlign: "center"
  }
})
export default SortMenu