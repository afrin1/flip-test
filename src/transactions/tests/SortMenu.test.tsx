import { render } from '@testing-library/react-native'
import React from 'react'
import SortMenu, { sortOptions } from '../SortMenu'

describe('SortMenu', () => {
  const setMenuCloseMock = jest.fn()
  const onOptionSelectionMock = jest.fn()

  it('should match snapshot', () => {
    expect(render(
      <SortMenu
        setMenuClose={setMenuCloseMock}
        onOptionSelection={onOptionSelectionMock}
        selectedOption={sortOptions.Nama_A_Z} />)
    ).toMatchSnapshot()
  })

  it(`Radio button should display the default value 
    as "Nama_A_Z"`, () => {
    const { getByTestId } = render(
      <SortMenu
        setMenuClose={setMenuCloseMock}
        onOptionSelection={onOptionSelectionMock}
        selectedOption={sortOptions.Nama_A_Z} />)

    expect(getByTestId('sort-radio-buttons'))
      .toHaveProp('defaultValue', sortOptions.Nama_A_Z)
  })

  it(`should have onOptionSelection callback set correctly`, () => {
    const { getByTestId } = render(
      <SortMenu
        setMenuClose={setMenuCloseMock}
        onOptionSelection={onOptionSelectionMock}
        selectedOption={sortOptions.Nama_A_Z} />)

        expect(getByTestId('sort-radio-buttons'))
        .toHaveProp('onSelection', onOptionSelectionMock)
  })
})