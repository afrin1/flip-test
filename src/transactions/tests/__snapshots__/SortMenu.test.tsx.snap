// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortMenu should match snapshot 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.5)",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "flexGrow": 1,
          "justifyContent": "center",
          "margin": 20,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <View
        defaultValue="Nama A-Z"
        items={
          Array [
            Object {
              "key": "URUTKAN",
              "text": "URUTKAN",
            },
            Object {
              "key": "Nama A-Z",
              "text": "Nama A-Z",
            },
            Object {
              "key": "Nama Z-A",
              "text": "Nama Z-A",
            },
            Object {
              "key": "Tanggal Terbaru",
              "text": "Tanggal Terbaru",
            },
            Object {
              "key": "Tanggal Terlama",
              "text": "Tanggal Terlama",
            },
          ]
        }
        onSelection={[MockFunction]}
        testID="sort-radio-buttons"
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#FF6347",
                  "borderRadius": 100,
                  "borderWidth": 2,
                  "height": 20,
                  "justifyContent": "center",
                  "width": 20,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 20,
                  "fontWeight": "700",
                  "marginLeft": 20,
                }
              }
            >
              URUTKAN
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#FF6347",
                  "borderRadius": 100,
                  "borderWidth": 2,
                  "height": 20,
                  "justifyContent": "center",
                  "width": 20,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#FF6347",
                    "borderRadius": 50,
                    "height": 10,
                    "width": 10,
                  }
                }
              />
            </View>
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 20,
                  "fontWeight": "700",
                  "marginLeft": 20,
                }
              }
            >
              Nama A-Z
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#FF6347",
                  "borderRadius": 100,
                  "borderWidth": 2,
                  "height": 20,
                  "justifyContent": "center",
                  "width": 20,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 20,
                  "fontWeight": "700",
                  "marginLeft": 20,
                }
              }
            >
              Nama Z-A
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#FF6347",
                  "borderRadius": 100,
                  "borderWidth": 2,
                  "height": 20,
                  "justifyContent": "center",
                  "width": 20,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 20,
                  "fontWeight": "700",
                  "marginLeft": 20,
                }
              }
            >
              Tanggal Terbaru
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#FF6347",
                  "borderRadius": 100,
                  "borderWidth": 2,
                  "height": 20,
                  "justifyContent": "center",
                  "width": 20,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 20,
                  "fontWeight": "700",
                  "marginLeft": 20,
                }
              }
            >
              Tanggal Terlama
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
